swaybar/tray/item.c

```diff
@@ -419,18 +427,16 @@ uint32_t render_sni(cairo_t *cairo, struct swaybar_output *output, double *x,
 				struct swaybar_pixmap *pixmap = pixmaps->items[idx];
 				cairo_surface_destroy(sni->icon);
 				sni->icon = cairo_image_surface_create_for_data(pixmap->pixels,
-						CAIRO_FORMAT_ARGB32, pixmap->size, pixmap->size,
-						cairo_format_stride_for_width(CAIRO_FORMAT_ARGB32, pixmap->size));
+						CAIRO_FORMAT_ARGB32, pixmap->width, pixmap->height,
+						cairo_format_stride_for_width(CAIRO_FORMAT_ARGB32, pixmap->width));
 			}
 		}
 	}
 
 	int icon_size;
 	cairo_surface_t *icon;
 	if (sni->icon) {
-		int actual_size = cairo_image_surface_get_height(sni->icon);
-		icon_size = actual_size < ideal_size ?
-			actual_size*(ideal_size/actual_size) : ideal_size;
+		icon_size = ideal_size;
```

Comment:
-1 on this change because it causes icons to look ugly when scaling only a little, say, from 20 to 22.

Reply:
Oh, right. Integer maths. I'll change it.